pkg/skaffold/build/cluster/cluster.go

```diff
@@ -42,7 +47,41 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, tags tag.ImageTags,
 		defer teardownDockerConfigSecret()
 	}
 
-	return build.InParallel(ctx, out, tags, artifacts, b.buildArtifactWithKaniko)
+	kanikoArtifacts, otherArtifacts := splitArtifacts(artifacts)
+
+	pArts, parallelErr := build.InParallel(ctx, out, tags, kanikoArtifacts, b.buildArtifactWithKaniko)
+	if parallelErr != nil {
+		logrus.Warnf("error building kaniko artifacts in parallel: %v", parallelErr)
+	}
+
+	sArts, err := build.InSequence(ctx, out, tags, otherArtifacts, b.runBuildForArtifact)
+	if err != nil {
+		return append(pArts, sArts...), err
+	}
+
+	return append(pArts, sArts...), parallelErr
+}
+
+func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, artifact *latest.Artifact, tag string) (string, error) {
+	switch {
+	case artifact.KanikoArtifact != nil:
+		return b.buildArtifactWithKaniko(ctx, out, artifact, tag)
+
+	case artifact.CustomArtifact != nil:
+		return b.buildArtifactWithCustomBuilder(ctx, out, artifact, tag)
+
```

Comment:
I see you added bazel artifact in the test cases. 
shd there be a case of bazelArtifact ? or something you are going to add later?

Reply:
I replaced bazelArtifact with customArtifact (guessing that's what you meant?), since bazel actually can't be run on cluster it's not really relevant here. 

Thanks for pointing that out!